sbol_utilities/default_components.py

```diff
@@ -0,0 +1,148 @@
+
+    deg_tag = sbol3.Component(name, sbol3.SBO_DNA)
+    deg_tag.name = name
+    deg_tag.description = description
+    deg_tag.roles. append(tyto.SO.protein_stability_element)
+    deg_tag.sequences.append(deg_tag_seq)
+    return tuple([deg_tag, deg_tag_seq])
+
+def plasmid_vector(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a plasmid vector or backbone and a sequence Components with a given name, description, and sequence.
+    """
+    plasmid_vector_seq = sbol3.Sequence(f'{name}_seq')
+    plasmid_vector_seq.elements= sequence
+    plasmid_vector_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    plasmid_vector = sbol3.Component(name, sbol3.SBO_DNA)
+    plasmid_vector.name = name
+    plasmid_vector.description = description
+    plasmid_vector.roles. append(tyto.SO.plasmid_vector)
+    plasmid_vector.sequences.append(plasmid_vector_seq)
+    return tuple([plasmid_vector, plasmid_vector_seq])
+
+def engineered_region(name: str, description: str, sub_components: List[sbol3.SubComponent])-> sbol3.Component:
+    """
+    Creates an engineered region Componentswith a given name, description, and sub components.
+    """
+    #sequence = blunt or assembly?
+    #engineered_region_seq = sbol3.Sequence(f'{name}_seq')
+    #engineered_region_seq.elements= sequence
+    #engineered_region_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    engineered_region = sbol3.Component(name, sbol3.SBO_DNA)
+    engineered_region.name = name
+    engineered_region.description = description
+    engineered_region.roles.append(sbol3.SO_ENGINEERED_REGION)
+    #create subcomponents if Components are passed  
+    for x in sub_components:
+        engineered_region.features.append(x)
+    #fix order
+    if len(engineered_region.features) > 1:
+            for i in range(len(engineered_region.features)-1):
+                engineered_region.constraints = [sbol3.Constraint(sbol3.SBOL_PRECEDES, engineered_region.features[i], engineered_region.features[i+1])]
+    else: pass
+    #engineered_region.sequences.append(engineered_region_seq)
+    return engineered_region # , engineered_region_seq
+
+def media(name: str, description: str)-> sbol3.Component:
+    """
+    Creates a media Component with a given name and description.
+    """
+    media = sbol3.Component(name, sbol3.SBO_FUNCTIONAL_ENTITY)
+    media.name = name
+    media.description = description
+    media.roles.append(tyto.NCIT.Media)
+    return media
+
+def strain(name: str, description: str)-> sbol3.Component:
+    """
+    Creates a strain Component with a given name and description.
+    """
+    strain = sbol3.Component(name, sbol3.SBO_FUNCTIONAL_ENTITY)
+    strain.name = name
+    strain.description = description
+    strain.roles.append(tyto.NCIT.Strain)
+    return strain
+
+def simple_chemical(name: str, description: str, chebi: str )-> sbol3.Component:
```

Comment:
ChEBI isn't the only way to make a simple chemical. There's also PubChem, SMILES, etc.
I'm also wondering if this would do better as a way to add a LocalSubComponent to a system Component, rather than stand-along definitions when we can avoid it.

Reply:
If I understand it correct, the best way to implement this is through the SEP054? If that is the case I agree that a conversion package is the best solution. How long could it take to be implemented? How can we use a function like this on the meantime?

Reply:
Above, I meant to say ExternallyDefined instead of LocalSubComponent.  Those don't need SEP_054 -- the whole point of ExternallyDefined is to allow us to do this with just an ontology term (which can be accessed via tyto), rather than needing to create a TopLevel.

Reply:
It could be like this:
```
def ed_simple_chemical(definition: str, **kwargs)-> sbol3.Component:
    """
    Creates an ExternallyDefined Simple Chemical Component.

    :param definition: The URI that links to a canonical definitino external to SBOL, recommended ChEBI.
    :param **kwargs: Keyword arguments of any other ExternallyDefined attribute.
    :return: A Component object.
    """
    ed_simple_chemical = sbol3.ExternallyDefined([sbol3.SBO_SIMPLE_CHEMICAL], definition, **kwargs)
    return ed_simple_chemical
```

Reply:
This looks reasonable. Would you consider providing the component that is to be its context to add directly into, like some of the existing functions do for things like interactions?

Reply:
Sorry I didn't understand that, could you provide an example?

Reply:
In component, check out routines like add_feature, constitutive, and add_interaction for somewhat similar cases.